[azure-core-cpp] Update to 1.1.0

azure-sdk opened the pull request:
Update vcpkg ports for Azure SDK release. This release may contain multiple ports.

azure-sdk committed:
[azure-core-cpp] Update to 1.1.0
## 1.1.0 (2021-07-02)

### Bugs Fixed

- Fixed a memory leak issue in `Base64Encode()`. (A community contribution, courtesy of _[jorgen](https://github.com/jorgen)_)

### Other Changes

- Made internal-only changes to support the Azure Key Vault client library.

### Acknowledgments

Thank you to our developer community members who helped to make Azure Core better with their contributions to this release:

- Jorgen Lind _([GitHub](https://github.com/jorgen))_

JonLiu1993 commented:
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

ahsonkhan commented:
cc @RickWinter, @vhvb1989

ahsonkhan commented:
cc @ras0219-msft

ras0219-msft commented on ports/azure-core-cpp/vcpkg.json:
```diff
@@ -37,7 +38,8 @@
           "features": [
             "ssl"
-          ]
+          ],
+          "version>=": "7.44"
```
This version constraint will fail if users are using the curated catalog, because there is no entry for `7.44` in the curl version database: https://github.com/microsoft/vcpkg/blob/master/versions/c-/curl.json

```suggestion
          ]
```

ahsonkhan replied:
Is it feasible for us to add a version entry for 7.44 to the curl version database?

I am curious to understand the rationale why it went down to 7.48 only:
https://github.com/microsoft/vcpkg/pull/15652

FYI @antkmsft, we'd want to make a similar change in https://github.com/Azure/azure-sdk-for-cpp/blob/a8d7a48a07a1eed8cc60dc00a7f618b6282c5eb7/sdk/core/azure-core/vcpkg/vcpkg.json#L44-L46

ras0219-msft replied:
As policy, we don't backfill older versions into the public catalog because there's no reasonable way for us to test them (and ensure they're fit for use). Since 7.48.0 is the oldest version available in the public catalog, the two reasonable approaches are:
1. Don't specify a constraint -- all users will have 7.48 or newer if they're using the curated catalog and if a user is trying to use a custom older version they'll just use overrides anyway to force that version
2. Specify a constraint on the oldest version _with a compatible version scheme_ in the curated catalog (7.74.0#5).

I strongly recommend option 1.

antkmsft replied:
@ahsonkhan, I'd do the following: apply this PR suggestion, and make a similar PR removing the very same line from vcpkg.json in our repo. No other work is necessary. I am glad it is known to us now, so we don't have to make a `port-version: 1` commit to vcpkg repo removing the version.

antkmsft replied:
I forgot to add, after we commit the version removal, we'll need to run `vcpkg x-add-version --overwrite-version azure-core-cpp` and do a subsequent commit.

antkmsft replied:
I made the updates I talked above

ahsonkhan replied:
Thanks for the fix @antkmsft - @ras0219 please let us know if it looks good now. Thanks!

azure-sdk committed:
[azure-storage-common-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

### Bug Fixes

- Fixed a memory leak issue while parsing XML.

azure-sdk committed:
[azure-storage-blobs-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

### Bug Fixes

- Fixed a bug where transactional MD5 hash was treated as blob MD5 hash when downloading partial blob.

azure-sdk committed:
[azure-storage-files-datalake-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

### Bug Fixes

- Fixed a bug where transactional MD5 hash was treated as blob MD5 hash when downloading partial blob.

azure-sdk committed:
[azure-storage-files-shares-cpp] Update to 12.0.1
## 12.0.1 (2021-07-07)

No API changes since `12.0.0`.

antkmsft committed:
Remove curl version from vcpkg manifest

antkmsft committed:
Merge branch 'azure-sdk-for-cpp-jul-2021' of https://github.com/azure-sdk/vcpkg into azure-sdk-for-cpp-jul-2021

antkmsft committed:
Update version files

ras0219-msft commented:
LGTM, thanks!
